File: Sorting/InsertionSort.java
import java.util.Scanner;

public class InsertionSort {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);
        int size;
        System.out.print("Enter array size : ");
        size = input.nextInt();
        int numbers[] = new int[size];

        for (int i = 0; i < size; i++) {
            numbers[i] = input.nextInt();
        }
        input.close();
        insertionSort(numbers, size);
        printArray(numbers, size);
    }

    public static void insertionSort(int num[], int len) {
        for (int i = 0; i < len; i++) {
            for (int j = i; j > 0; j--) {
                if (num[j - 1] > num[j]) {
                    int temp = num[j - 1];
                    num[j - 1] = num[j];
                    num[j] = temp;
                }
            }
        }
    }

    public static void printArray(int num[], int len) {
        for (int i = 0; i < len; i++) {
            System.out.print(num[i] + " ");
        }
    }
}

File: Sorting/InsertionSort.py
def insertionSort(num):
    for i in range(len(num)):
        for j in range (i,0,-1):
            if num[j-1]>num[j]:
                temp=num[j-1];
                num[j-1]=num[j];
                num[j]=temp;
                
numbers=input() 
 
numbers=list(numbers.split())
numbers=[int(x) for x in numbers]

insertionSort(numbers)
print(numbers)